algorithms/binary-search.py

```diff
@@ -0,0 +1,88 @@
+def bin_search_recursive(arr, val, left=0, right=None):
```

Comment:
Obviously this one I don't like. it's too long, too many `if`s inside, hard to understand what going on...